examples/gke-cloud-connector-ts/index.ts

```diff
@@ -0,0 +1,147 @@
+// Copyright 2016-2021, Pulumi Corporation.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as google from "@pulumi/google-native";
+import * as random from "@pulumi/random";
+import * as k8s from "@pulumi/kubernetes";
+
+const config = new pulumi.Config("google-native");
+const project = config.require("project");
+
+// TODO: Determine this dynamically once https://github.com/pulumi/pulumi-google-native/issues/166 is done.
+const engineVersion = "1.19.9-gke.1900";
+
+const nodeConfig: google.types.input.container.v1.NodeConfigArgs = {
+    machineType: "n1-standard-2",
+    oauthScopes: [
+        "https://www.googleapis.com/auth/compute",
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring"
+    ],
+    preemptible: true,
+};
+
+// Create a GKE cluster with the config connector addon enabled.
+const cluster = new google.container.v1.Cluster("cluster", {
+    initialClusterVersion: engineVersion,
+    nodePools: [{
+        config: nodeConfig,
+        initialNodeCount: 1,
+        management: {
+            autoRepair: false,
+        },
+        name: "initial",
+    }],
+    addonsConfig: {
+        configConnectorConfig: {
+            enabled: true
+        }
+    },
+    loggingService: "logging.googleapis.com/kubernetes", // stackdriver
+    workloadIdentityConfig: {
+        workloadPool: `${project}.svc.id.goog`
+    },
+});
+
+// Create a kubeconfig to connect to the GKE cluster.
+const kubeconfig = pulumi.all([cluster.name, cluster.endpoint, cluster.masterAuth]).apply(
+    ([name, endpoint, auth]) => {
+        const context = `${project}_${name}`;
+        return `apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: ${auth.clusterCaCertificate}
+    server: https://${endpoint}
+  name: ${context}
+contexts:
+- context:
+    cluster: ${context}
+    user: ${context}
+  name: ${context}
+current-context: ${context}
+kind: Config
+preferences: {}
+users:
+- name: ${context}
+  user:
+    auth-provider:
+      config:
+        cmd-args: config config-helper --format=json
+        cmd-path: gcloud
+        expiry-key: '{.credential.token_expiry}'
+        token-key: '{.credential.access_token}'
+      name: gcp
+`;
+});
+
+// Create a Provider to use the new kubeconfig.
+const k8sProvider = new k8s.Provider("k8s", {kubeconfig});
+
+// Create a random suffix for the ServiceAccount.
+const saSuffix = new random.RandomPet("saSuffix", {length: 1}).id;
```

Comment:
Did you intend for this to be a RandomString instead?

Reply:
No, I used pet because it was a little easier to find in the console when I was poking around.